cql3/restrictions/single_column_primary_key_restrictions.hh

```diff
@@ -307,7 +307,12 @@ private:
     }
 
 public:
-    std::vector<bounds_range_type> bounds_ranges(const query_options& options) const override;
+    // When ValueType is clustering_key_prefix, this overrides a method of clustering_key_restrictions.
+    // When ValueType is partition_key, this method is unused. That's why we can't add `override`
+    // to the declaration.
+    // It would be great if we could conditionally implement this method only when
+    // ValueType is partition_key, but this seems to be impossible in C++.
```

Comment:
Did you mean `when ValueType is clustering_key_prefix`?

Reply:
Yes, that's what I meant. Thanks. Fixed.